NitroxClient/Communication/LANDiscoveryClient.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using LiteNetLib;
+using LiteNetLib.Utils;
+using NitroxModel.Constants;
+using NitroxModel.Logger;
+
+namespace NitroxClient.Communication
+{
+    public static class LANDiscoveryClient
+    {
+        private static readonly List<IPEndPoint> discoveredServers = new();
+
+        private static Action<IPEndPoint> foundServerCallback;
+
+        private static EventBasedNetListener listener;
+        private static NetManager client;
+
+        private static Timer broadcastTimer;
+        private static Timer pollTimer;
+
+        public static void SearchForServers(Action<IPEndPoint> callback)
```

Comment:
You never stop the discovery client, right? At least after joining LAN-Discovery should be disabled.

Reply:
I recommend turning this into an async task function with cancellation token parameter. Cancellation tokens can have a timeout.

Reply:
Why would using an async function have an advantage here? Couldn't I just make a ServerFound event and fire it whenever a server is detected?

Reply:
It would be if you didn't use LiteNetLib to starts a new server but instead used UDP broadcast. LiteNetLib does not have async/await support (only event based).